packages/core/src/call-optional-signature.ts

```diff
@@ -0,0 +1,22 @@
+import { keccak256 } from 'ethereumjs-util';
+import { call, EthereumProvider } from './provider';
+
+export async function callOptionalSignature(provider: EthereumProvider, address: string, signature: string) {
```

Comment:
This function is limited to function with no arguments, and the output is not parsed. That is good enough for now, but if we ever want to go further, I believe that ethers has some usefull primitives that are ts friendly.

Reply:
Currently the `upgrades-core` package currently does not pull in ethers as a dependency, we can consider it when there is a need.